app/use_cases/petition_plugin/detail_updater.rb

```diff
@@ -0,0 +1,14 @@
+module PetitionPlugin
+  class DetailUpdater
+    def perform(detail, attrs, version_attrs)
+      detail.update_attributes attrs 
+
+      current_version = detail.current_version
+      if current_version.nil? || current_version.body != version_attrs[:body]
+        detail.petition_detail_versions << PetitionPlugin::DetailVersion.new(version_attrs)
+      end
+
+      detail.save
```

Comment:
Yes there are validations, by they somehow are automatically been sent to the client when errors happen, on the controller I don't use the `detail.errors` or even the  `detail` instance after this method is called.

Anyway, what would be your suggestion? Return the `detail` object intead, and verify if there are errors?